### ISSUE TITLE
Feature Request/Question: Resume from specific epoch/step? (And is manually editing `latest` safe?)

### ISSUE THREAD

**Issue by smn3786** [tdrussell/diffusion-pipe#263]
Hello.
First off, massive thanks for this awesome project!

I've been training loras and had a couple of q's about resuming.

Resuming from the latest checkpoint with `--resume_from_checkpoint` works great. But what if I want to go back to an earlier epoch (say, epoch 40 out of 65) and continue with a different learning rate?

1.  Is there a way to directly resume from a specific epoch folder (like `20250511_05-04-23/epoch40`) or a specific global step?
    Tried pointing `--resume_from_checkpoint` to an epoch folder directly, but no dice. Would be super handy if this was possible for trying out different LR schedules manually. Any plans for something like this?

2.  As a workaround, I edited the `latest` file in the checkpoint folder to point to `global_step723` and then resumed. It seemed to work!
    **Question is: is this a safe hack?** Or could it mess with things like the optimizer state or epoch counts down the line? Just want to make sure I'm not accidentally breaking stuff.

**Comment by tdrussell** [comment]
(2) is the only way to do this right now, and yes it is safe. Deepspeed just looks in the "latest" file to find which checkpoint folder to resume from. You can change that file freely as long as the checkpoint folder exists.

You can also restart training entirely, but using existing lora weights as initialization, using the init_from_existing option. But this way doesn't preserve optimizer states.

**Comment by smn3786** [comment]
@tdrussell, Thanks for the answer!

Gotta admit, I felt a bit silly after opening this issue.

So, I checked out the `training.toml` file that sets up the training, and just like you said, there was a comment about `init_from_existing`.  
I figured out that if I uncomment it and just drop in the epoch path I need, like this:

```toml
init_from_existing = 'my/wan/training-data/output/20250512_03-34-22/epoch15'
```

...I can start a new training session.

For now, this method totally nails what I need, and I've learned it's actually better for keeping things tidy since it's managed as its own separate space, kinda like making a new branch, which is pretty sweet.

Cheers,

**Comment by alienzed** [comment]
> Hi! maybe it's not good to ask in existing issue, but I think it's related somehow. Can I change 'checkpoint_every_n_minutes' value between training sessions in one project? I mean - can I change this setting alone in existing setup after interrupting a session and before resuming it?

Yes, that works very well. And I hope it's not rude to say that **yes you should open new issues for new questions**.

After reading the config of course, let's let tdrussel focus on the nitty gritty and not have to answer questions his excellent examples already answer ;)
☮️
